framework/helpers/BaseStringHelper.php

```diff
@@ -104,12 +104,15 @@ public static function dirname($path)
      */
     public static function truncate($string, $length, $suffix = '...', $encoding = null, $asHtml = false)
     {
+        if ($encoding === null) {
+            $encoding = Yii::$app ? Yii::$app->charset : 'UTF-8';
```

Comment:
> For the helper it is no problem as long as you specify the $encoding param of the method.

You can do it now, with current code, and thus avoid coupling with `Application`. Following such argument - no change is needed at all.